commit bbdbbd16a7aa0293e862c759b7808546d453d6bd
Author: Alex Wilson <alex@uq.edu.au>
Date:   2019-08-27T23:00:23+00:00 (6 weeks ago)
    
    joyent/node-triton#265 Un-hide --act-as option and add environment var
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Reviewed by: Trent Mick <trent.mick@joyent.com>
    Approved by: Trent Mick <trent.mick@joyent.com>

diff --git a/lib/cli.js b/lib/cli.js
index ca6acd0..43ca473 100644
--- a/lib/cli.js
+++ b/lib/cli.js
@@ -88,11 +88,11 @@ var OPTIONS = [
     {
         names: ['act-as'],
         type: 'string',
-        help: 'Masquerade as the given account login name. This can only ' +
-            'succeed for operator accounts. Note that accesses like these ' +
-            'audited on the CloudAPI server side.',
-        helpArg: 'ACCOUNT',
-        hidden: true
+        help: 'Masquerade as the given account login name. This is useful ' +
+            'for operator accounts and members of cross-account roles. Note ' +
+            'that accesses like these are audited on the CloudAPI server ' +
+            'side. Environment: TRITON_ACT_AS=ACCOUNT.',
+        helpArg: 'ACCOUNT'
     },
     {
         names: ['user', 'u'],
diff --git a/lib/config.js b/lib/config.js
index b5708b2..ee742b8 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -283,6 +283,10 @@ function _loadEnvProfile(profileOverrides) {
     }
     envProfile.url = process.env.TRITON_URL || process.env.SDC_URL;
     envProfile.keyId = process.env.TRITON_KEY_ID || process.env.SDC_KEY_ID;
+    var actAs = process.env.TRITON_ACT_AS;
+    if (actAs) {
+        envProfile.actAsAccount = actAs;
+    }
 
     if (process.env.TRITON_TLS_INSECURE) {
         envProfile.insecure = common.boolFromString(
